subgraph/src/indexes.ts

```diff
@@ -111,15 +111,93 @@ const computeUpdatedMorphoIndex = (
   return newIndex;
 };
 
+const computeUpdatedMorphoIndexV2 = (
+  marketAddress: Address,
+  blockTimestamp: BigInt,
+  lastMorphoIndex: BigInt,
+  lastUpdateBlockTimestamp: BigInt,
+  lastPercentSpeed: BigInt,
+  lastTotalScaled: BigInt,
+  marketSide: string
+): BigInt => {
+  const obj = ipfsJson();
+  // sync eventual previous epoch
+  const prevEpochId = timestampToEpochId(obj, lastUpdateBlockTimestamp);
+  const currentEpochId = timestampToEpochId(obj, blockTimestamp);
+  if (!currentEpochId) return lastMorphoIndex; // TODO: to move at the end
+
+  if (!prevEpochId && currentEpochId) {
+    // start of the first epoch
+    const start = epochIdToStartTimestamp(obj, currentEpochId);
+    if (!start) {
+      log.critical("No start timestamp for epoch {}", [currentEpochId as string]);
+      return BigInt.zero();
+    }
+    const distribution = fetchDistribution(obj, blockTimestamp, marketSide, marketAddress);
+    // consider the speed between onPool & inP2P to be constant between 2 markets update.
+    // That means that we consider that the difference of interest generated does not make changes between
+    // on pool & in P2P repartition, i.e. lastPercentSpeed is constant.
+    const speed = distribution.times(lastPercentSpeed).div(BigInt.fromString("10000"));
+    const accrualIndex = computeOneEpochDistribuedRewards(start, blockTimestamp, lastTotalScaled, speed);
+    return lastMorphoIndex.plus(accrualIndex);
+  }
+
+  if (
+    prevEpochId &&
+    currentEpochId &&
+    // string comparison is not working when compiled with WASM, we have to pass through bytes comparison
+    !Bytes.fromUTF8(prevEpochId.toString()).equals(Bytes.fromUTF8(currentEpochId.toString()))
```

Comment:
`prevEpochId` & `currentEpochId` are `Bytes` already, aren't they? It's not possible to compare them directly?

Reply:
Oh yeah youre right we are doing Bytes => String => Bytes